wcfsetup/install/files/ts/WoltLabSuite/Core/Acp/Ui/Menu/Item/Handler.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * Provides the interface logic to add and edit menu items.
+ *
+ * @author  Alexander Ebert
+ * @copyright  2001-2019 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Acp/Ui/Menu/Item/Handler
+ */
+
+import Dictionary from "../../../../Dictionary";
+import DomUtil from "../../../../Dom/Util";
+import * as Language from "../../../../Language";
+import * as UiPageSearchHandler from "../../../../Ui/Page/Search/Handler";
+
+class AcpUiMenuItemHandler {
+  private activePageId = 0;
+  private readonly cache = new Map<number, number>();
+  private readonly containerExternalLink: HTMLElement;
+  private readonly containerInternalLink: HTMLElement;
+  private readonly containerPageObjectId: HTMLElement;
+  private readonly handlers: Map<number, string>;
+  private readonly pageId: HTMLSelectElement;
+  private readonly pageObjectId: HTMLInputElement;
+
+  /**
+   * Initializes the interface logic.
+   */
+  constructor(handlers: Map<number, string>) {
+    this.handlers = handlers;
+
+    this.containerInternalLink = document.getElementById("pageIDContainer")!;
+    this.containerExternalLink = document.getElementById("externalURLContainer")!;
+    this.containerPageObjectId = document.getElementById("pageObjectIDContainer")!;
+
+    if (this.handlers.size) {
+      this.pageId = document.getElementById("pageID") as HTMLSelectElement;
+      this.pageId.addEventListener("change", this.togglePageId.bind(this));
+
+      this.pageObjectId = document.getElementById("pageObjectID") as HTMLInputElement;
+
+      this.activePageId = ~~this.pageId.value;
+      if (this.activePageId && this.handlers.has(this.activePageId)) {
+        this.cache.set(this.activePageId, ~~this.pageObjectId.value);
+      }
+
+      const searchButton = document.getElementById("searchPageObjectID")!;
+      searchButton.addEventListener("click", (ev) => this.openSearch(ev));
+
+      // toggle page object id container on init
+      if (this.handlers.has(~~this.pageId.value)) {
+        DomUtil.show(this.containerPageObjectId);
+      }
+    }
+
+    document.querySelectorAll('input[name="isInternalLink"]').forEach((input: HTMLInputElement) => {
+      input.addEventListener("change", () => this.toggleIsInternalLink(input.value));
+
+      if (input.checked) {
+        this.toggleIsInternalLink(input.value);
+      }
+    });
+  }
+
+  /**
+   * Toggles between the interface for internal and external links.
+   */
+  private toggleIsInternalLink(value: string): void {
+    if (~~value) {
+      DomUtil.show(this.containerInternalLink);
+      DomUtil.hide(this.containerExternalLink);
+      if (this.handlers.size) this.togglePageId();
```

Comment:
```suggestion
      if (this.handlers.size) {
        this.togglePageId();
      }
```

I wonder why Prettier did not force the braces here.

Reply:
The `.editorconfig` has conflicting settings that cause PhpStorm to keep them.